internal/configs/oidc/oidc.conf

```diff
@@ -0,0 +1,96 @@
+    }
+
+    location @do_oidc_flow {
+        status_zone "OIDC start";
+        js_content oidc.auth;
+        default_type text/plain; # In case we throw an error
+    }
+
+    #set $redir_location "/_codexch";
+    location = /_codexch {
+        # This location is called by the IdP after successful authentication
+        status_zone "OIDC code exchange";
+        js_content oidc.codeExchange;
+        error_page 500 502 504 @oidc_error;
+    }
+
+    location = /_token {
+        # This location is called by oidcCodeExchange(). We use the proxy_ directives
+        # to construct the OpenID Connect token request, as per:
+        #  http://openid.net/specs/openid-connect-core-1_0.html#TokenRequest
+        internal;
+        proxy_ssl_server_name on; # For SNI to the IdP
+        proxy_set_header      Content-Type "application/x-www-form-urlencoded";
+        proxy_set_body        "grant_type=authorization_code&client_id=$oidc_client&$args&redirect_uri=$redirect_base$redir_location";
+        proxy_method          POST;
+        proxy_pass            $oidc_token_endpoint;
+   }
+
+    location = /_refresh {
+        # This location is called by oidcAuth() when performing a token refresh. We
+        # use the proxy_ directives to construct the OpenID Connect token request, as per:
+        #  https://openid.net/specs/openid-connect-core-1_0.html#RefreshingAccessToken
+        internal;
+        proxy_ssl_server_name on; # For SNI to the IdP
+        proxy_set_header      Content-Type "application/x-www-form-urlencoded";
+        proxy_set_body        "grant_type=refresh_token&refresh_token=$arg_token&client_id=$oidc_client&client_secret=$oidc_client_secret";
+        proxy_method          POST;
+        proxy_pass            $oidc_token_endpoint;
+    }
+
+    location = /_id_token_validation {
+        # This location is called by oidcCodeExchange() and oidcRefreshRequest(). We use
+        # the auth_jwt_module to validate the OpenID Connect token response, as per:
+        #  https://openid.net/specs/openid-connect-core-1_0.html#IDTokenValidation
+        internal;
+        auth_jwt "" token=$arg_token;
+        js_content oidc.validateIdToken;
+        error_page 500 502 504 @oidc_error;
+    }
+
+    location = /logout {
+        status_zone "OIDC logout";
+        add_header Set-Cookie "auth_token=; $oidc_cookie_flags"; # Send empty cookie
+        add_header Set-Cookie "auth_redir=; $oidc_cookie_flags"; # Erase original cookie
+        js_content oidc.logout;
+    }
+
+    location = /_logout {
+        # This location is the default value of $oidc_logout_redirect (in case it wasn't configured)
+        default_type text/plain;
+        return 200 "Logged out\n";
+    }
+
+    location @oidc_error {
+        # This location is called when oidcAuth() or oidcCodeExchange() returns an error
+        status_zone "OIDC error";
+        default_type text/plain;
+        return 500 $internal_error_message;
+    }
+
+    # location /api/ {
```

Comment:
Should this commented section be removed?

Reply:
I think this was intentionally left so it is the only modification made to https://github.com/nginxinc/nginx-openid-connect/blob/master/openid_connect.server_conf . 

This way, if we want to apply updates to a new version of the file from that repo: the fact that `location /api/` should be removed should be clear from the diff.

Reply:
Ah makes sense, thanks.

Reply:
yep, that was the idea 🙂